crates/sncast/src/starknet_commands/script.rs

```diff
@@ -34,21 +34,83 @@ use itertools::chain;
 use num_traits::ToPrimitive;
 use runtime::EnhancedHintError;
 use scarb_artifacts::ScarbCommand;
+use serde::Serialize;
 use sncast::helpers::response_structs::ScriptResponse;
 use sncast::helpers::scarb_utils::{
     get_package_metadata, get_scarb_manifest, get_scarb_metadata_with_deps, CastConfig,
 };
+use sncast::{print_formatted, stringify_command_result_struct, ValueFormat};
 use starknet::accounts::Account;
 use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
 use starknet::providers::jsonrpc::HttpTransport;
 use starknet::providers::JsonRpcClient;
 use tokio::runtime::Runtime;
 
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Verbosity {
+    Quiet,
+
+    #[default]
+    Normal,
+}
+
+#[derive(Debug)]
+pub struct UI {
```

Comment:
This probably should be unified with `lib::print_command_result` at least partailly? I suggest looking on https://github.com/foundry-rs/starknet-foundry/pull/1397

We need to think about stuff like unifying numbers formatting etc.

Reply:
extracted some common logic

Reply:
done